libr/anal/p/anal_x86_cs.c

```diff
@@ -76,31 +77,34 @@ static bool is_xmm_reg(cs_x86_op op) {
  * @param  n       Operand index
  * @param  set     if 1 it adds set (=) to the operand
  * @param  setoper Extra operation for the set (^, -, +, etc...)
- * @return         char* with the esil operand
+ * @param  out     Output
+ * @return         Void
  */
-static char *getarg(struct Getarg* gop, int n, int set, char *setop) {
+static void getarg(struct Getarg* gop, int n, int set, char *setop, char *out) {
```

Comment:
uhm i think is better to have an index to point to a static char buf[4][64]; this way we can drop the char *out parameter and keep using the char *src = getarg() thing with less repeated code. The call to getarg() must specify the buffer index it will use to store the value. Tihs way we also avoid using the heap and we can return "" instead of NULL which is always nicer to print